packages/core/src/indexing/__tests__/local-index-api.test.ts

```diff
@@ -48,6 +49,49 @@ describe('with database backend', () => {
     })
     expect(response.edges.map((e) => e.node.content)).toEqual(backendPage.edges.map((e) => e.node))
   })
+
+  test('return empty page from the database if cannot retrieve stream state', async () => {
+    const query = { model: 'foo', first: 1 }
+    const backendPage: Page<string> = {
+      edges: [
+        {
+          cursor: randomString(32),
+          node: null,
+        }
+      ],
+      pageInfo: {
+        hasPreviousPage: false,
+        hasNextPage: false,
+        startCursor: 'startCursor',
+        endCursor: 'endCursor',
+      },
+    }
+    const pageFn = jest.fn(async () => backendPage)
+    const streamStateFn = jest.fn(async (streamId: any) => {
+      return undefined
+    })
+    const fauxBackend = { page: pageFn } as unknown as DatabaseIndexApi
+    const fauxRepository = { streamState: streamStateFn } as unknown as Repository
+    const fauxLogger = {
```

Comment:
we should check if the message is logged. one more expect statement to check if `warn` is called , and it is ok by me